### DISCUSSION TITLE
Кодирование пола пациента

### DISCUSSION THREAD

**Discussion by niquola** [fhir-ru/core#34] [Административная РГ] [answered]
Как мы будем кодировать пол пациента и гармонизировать с НСИ справочником - ConceptMap?

* FHIR ValueSet - https://www.hl7.org/fhir/valueset-administrative-gender.html

Предыдущее обсуждение - https://github.com/fhir-ru/core/issues/12

**Comment by iradio** [comment]
Для наглядности, сейчас соответствие представляется следующим образом
| [FHIR ValueSet](https://www.hl7.org/fhir/valueset-administrative-gender.html) | [НСИ Минздрава](https://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.11.1040/version/2.1) | [НСИ ФОМС](http://nsi.ffoms.ru/#refbookList?refbookList:filter:$filter=set&refbookList:filter:name=%D0%BF%D0%BE%D0%BB&refbookList:$active=1&refbookList:$selectedId=914/refbookList.refbookList.view$2v914?menu:filter:dictionaryId=914&menu:filter:readOnly=true&menu:$active=1&menu:$selectedId=undefined&refbookSimple0:$active=1&refbookSimple0:page=1&refbookSimple1:$active=1&refbookSimple1:page=1&refbookSimple1:$selectedId=undefined) | 
| --- | ----------- | ----------- |
| Male | Мужской | Мужской  |
| Female | Женский | Женский |
| Other | - | - |
| Unknown | Неопределенный  | - |

**Comment by Vrodvi** [comment]
Вопрос пересекается с https://github.com/fhir-ru/core/discussions/78.

Мне кажется самым универсальным вариантом - использованный в голландском проекте (через extension).
Просто ConceptMap не позволит указать код по НСИ в самом экземпляре ресурса, а это ожидается.

В данном случае может быть так:
```
    "gender": "female",
    "_gender": {
        "extension": [
            {
                "url": "http://fhir.ru/fhir/Extension/gender-ext-codeList",
                "CodeableConcept": [
                    {
                        "system": "https://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.11.1040",
                        "version": "2.1",
                        "code": "2",
                        "display": "Женский"
                    }
                ]
            }
        ]
    }
```

Или вместо `"url": "http://fhir.ru/fhir/Extension/gender-ext-codeList"` сделать одно универсальное расширение, которое можно будет использовать для разных атрибутов - `"url": "http://fhir.ru/fhir/Extension/uv-ext-codeList"`.

**Comment by eugenekogan-spb** [comment, marked as answer]
В обсуждении http://fhir-ru.zendoc.me/meetings.2022-03-29 предварительное решение - создать code system для НСИ МЗ РФ, прописать маппинг - concept map, как описано выше. При этом можно не делать extension и русскоязычное наименование пола в сам ресурс не добавлять.

**Comment by Vrodvi** [comment]
@niquola @eugenekogan-spb Мне кажется вы преждевременно закрыли данное обсуждение:
- ConceptMap'а ещё нет
- чтобы составить ConceptMap'а нужно предварительно определить набор значений НСИ'шного справочника (т.е. определить ресурсы CodeSystem и ValuSet) - это также не сделано
- каким образом будут определяться CodeSystem и ValuSet НСИ'шных справочников также не решено, плюс решили не вести терминологический сервер, который бы помог в этом
- если русскоязычное наименование не будет добавляться в ресурс, то нужно подготовить чёткий алгоритм получения русскоязычного наименования из ConceptMap'а (а точнее - кода из ConceptMap'а, а затем по этому коду - значения из CodeSystem, которого ещё нет - см. выше)

**Reply by eugenekogan-spb** [reply]
Полностью согласен. Моей задачей было - разделить дискуссии, по которым решение принято, хотя бы предварительно (и следует отложить обсуждение пока это решение не реализовано), и дискуссии, где никакого решения пока нет. Как это лучше сделать?

**Reply by Vrodvi** [reply]
Можно использовать механизм меток - https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#applying-a-label. 
Добавить метку вроде [есть предварительное решение] \ [в работе] \ [обрабатывается].
Метки уже ставятся на Issues, их также можно ставить и на Дискуссии. 
Но метки могут ставить только  Contributor'ы.

**Reply by Vrodvi** [reply]
@eugenekogan-spb 

Вот этот замечание не решается простым определением ConceptMap'а:
> если русскоязычное наименование не будет добавляться в ресурс, то нужно подготовить чёткий алгоритм получения русскоязычного наименования из ConceptMap'а (а точнее - кода из ConceptMap'а, а затем по этому коду - значения из CodeSystem, которого ещё нет - см. выше)

Вот тут http://fhir-ru.zendoc.me/systems.rosminzdrav-gender#:concept-map завели ConceptMap.
ConceptMap'ы в FHIR однонаправленные - в данном случае направление перекодирования из НСИ в FHIR.

Есть вопросы:
1) Как это предполагается использовать?
Данный ConceptMap предназначается для систем, работающие с кодами НСИ, чтобы они могли перед формированием ресурса конвертировать коды НСИ в коды FHIR, правильно я понял?

2) Что делать системам, которые будут разбирать ресурсы, как они должны получить коды НСИ на основании FHIR кодов, указанных в ресурсе?

3) Что делать если в ресурсе указан FHIR-код, которого нет в ConceptMap, например "Other"? 
Нужно либо запрещать использование кода "Other", либо делать ConceptMap в другую сторону - от FHIR к НСИ.
Чтобы код "Other" не использовался нужно определять сокращённый набор значений для FHIR-кодов пола пациента, но я не вижу в профиле пациента (http://fhir-ru.zendoc.me/core.Patient#:profile) такого сокращённого набора значений для пол.

**Reply by eugenekogan-spb** [reply]
Решение на созвоне 2022-04-08: делаем маппинг в обе стороны, тексте IG рекомендуем не использовать Other и объясним почему. В профиле средствами Zen запретим использование этого кода.

**Reply by Vrodvi** [reply]
> В тексте IG рекомендуем не использовать Other 

Почему только в тексте, если для этого есть формализованный механизм определения наборов значений?

Чтобы код "Other" не использовался нужно определять сокращённый набор значений для FHIR-кодов пола пациента, нужно определить ограниченный набор значений и прописать его в профиле пациента:
```
{
  "resourceType": "ValueSet",
  "id": "core-vs-administrative-gender",
  "name": "CoreVsAdministrativeGender",
  "title": "Core ValueSet Пол пациента (FHIR AdministrativeGender)",
  "url" : "http://fhir.ru/core/vs/core-vs-administrative-gender",
  "status": "active",
  "compose": {
    "include": [
      {
        "system": "http://hl7.org/fhir/administrative-gende",
        "concept": [
          {
            "code": "male"
          },
          {
            "code": "female"
          },
          {
            "code": "unknown"
          }
        ]
      }
    ]
  }
}
```

Такой подход правда означает что и маппинги тоже нужно переделать - маппинг должен быть между НСИ-шным набором значений и данным ограниченным набор значений из FHIR.

---

Прошу прощения за некоторый нездоровый формализм в этом вопросе, он связан с тем, что хочется сформировать чёткие принципы по работе с наборами значений для использования в последующих профилях. 
Если даже для одного простого атрибута в виде пола пациента у нас уже образуется некоторая неоднозначность и недосказанность (часть требований формализована, а часть просто в тексте IG), то возникают опасения за все последующие профили, которые будут иметь существенно более сложную структуру.
